src/Sparrow/Utils/NativeMemory.cs

```diff
@@ -170,25 +187,25 @@ public static void DecrementLuceneManagedAllocations(long size)
         public static byte* AllocateMemoryForLuceneTermCache(long size)
         {
             Interlocked.Add(ref _totalLuceneUnmanagedAllocationsForTermCache, size);
-            return AllocateMemory(size, out _);
+            return AllocateMemory(size);
         }
 
         public static byte* AllocateMemoryForLuceneSorting(long size)
         {
             Interlocked.Add(ref _totalLuceneUnmanagedAllocationsForSorting, size);
-            return AllocateMemory(size, out _);
+            return AllocateMemory(size);
```

Comment:
You have modified `byte* AllocateMemory(long size)` method and started to use it here and above. Although there were already some usage of it. All of them will no longer update thread stats in all that cases. Is it what we want there as well? Did you check all usages?

Reply:
Yes, I checked all usages.
All of them are for static usage, which means we aren't going to release that memory, and it shouldn't be under thread stats since that makes it confusing.

`SmallStringCompression`
https://github.com/ravendb/ravendb/blob/45fc13508886c92e40f86b2dff6fc68eef4f077a/src/Sparrow/Compression/SmallStringCompression.cs#L68

`LazyStringParser`
https://github.com/ravendb/ravendb/blob/45fc13508886c92e40f86b2dff6fc68eef4f077a/src/Sparrow/Json/LazyStringParser.cs#L74

`GetNullValueUnmanagedString`
https://github.com/ravendb/ravendb/blob/45fc13508886c92e40f86b2dff6fc68eef4f077a/src/Raven.Server/Documents/Queries/Sorting/AlphaNumeric/AlphaNumericFieldComparator.cs#L27

All other usages are for tests.